providers/slack/resources/conversations.go

```diff
@@ -0,0 +1,175 @@
+package resources
+
+import (
+	"time"
+
+	"go.mondoo.com/cnquery/providers-sdk/v1/util/convert"
+
+	"go.mondoo.com/cnquery/providers/slack/connection"
+
+	"go.mondoo.com/cnquery/llx"
+
+	"go.mondoo.com/cnquery/providers-sdk/v1/plugin"
+
+	"github.com/slack-go/slack"
+)
+
+func (s *mqlSlack) conversations() ([]interface{}, error) {
+	conn := s.MqlRuntime.Connection.(*connection.SlackConnection)
+	client := conn.Client()
+
+	list := []interface{}{}
+
+	// https://api.slack.com/methods/conversations.list
+	// scopes: channels:read, groups:read, im:read, mpim:read
+	opts := &slack.GetConversationsParameters{
+		Limit: 1000, // use maximum
+		Types: []string{"public_channel", "private_channel", "mpim", "im"},
+	}
+
+	for {
+		conversations, cursor, err := client.GetConversations(opts)
+		if err != nil {
+			return nil, err
+		}
+		for i := range conversations {
+			mqlUser, err := newMqlSlackConversation(s.MqlRuntime, conversations[i])
+			if err != nil {
+				return nil, err
+			}
+			list = append(list, mqlUser)
+		}
+		// check if we are at the end of pagination
+		if cursor == "" {
+			break
+		}
+		opts.Cursor = cursor
+	}
+
+	return list, nil
+}
+
+// custom object to make sure the json values match and the time is properly parsed
+type topic struct {
+	Value   string     `json:"value"`
+	Creator string     `json:"creator"`
+	LastSet *time.Time `json:"lastSet"`
+}
+
+func newTopic(t slack.Topic) topic {
+	lastSet := t.LastSet.Time()
+	return topic{
+		Value:   t.Value,
+		Creator: t.Creator,
+		LastSet: &lastSet,
+	}
+}
+
+// custom object to make sure the json values match and the time is properly parsed
+
+type purpose struct {
+	Value   string     `json:"value"`
+	Creator string     `json:"creator"`
+	LastSet *time.Time `json:"lastSet"`
+}
+
+func newPurpose(p slack.Purpose) purpose {
+	lastSet := p.LastSet.Time()
+	return purpose{
+		Value:   p.Value,
+		Creator: p.Creator,
+		LastSet: &lastSet,
+	}
+}
+
+func newMqlSlackConversation(runtime *plugin.Runtime, conversation slack.Channel) (interface{}, error) {
+	topic, err := convert.JsonToDict(newTopic(conversation.Topic))
+	if err != nil {
+		return nil, err
+	}
+
+	purpose, err := convert.JsonToDict(newPurpose(conversation.Purpose))
+	if err != nil {
+		return nil, err
+	}
+
+	created := conversation.Created.Time()
+
+	var creator plugin.Resource
+
+	if conversation.Creator != "" {
+		creator, err = CreateResource(runtime, "slack.user", map[string]*llx.RawData{
```

Comment:
This won't work. I just ran into the same problem. `CreateResource` is used for creating new resources with pre-processed data. If you want to initialize a resource with "user" input and effectively call the `Init` function, you need tou use `NewResource`